spring-integration-core/src/test/java/org/springframework/integration/config/annotation/AnnotatedEndpointActivationTests.java

```diff
@@ -82,6 +93,15 @@ public void sendAndReceive() {
 		assertTrue(this.applicationContext.containsBean("annotatedEndpoint2.process.serviceActivator"));
 	}
 
+	@Test
+	public void sendAndReceiveAsync() {
+		this.inputAsync.send(new GenericMessage<String>("foo"));
+		Message<?> message = this.outputAsync.receive(100);
+		assertNotNull(message);
+		assertEquals("foo", message.getPayload());
+		assertTrue(this.applicationContext.containsBean("annotatedEndpoint3.process.serviceActivator"));
```

Comment:
If code in the beginning of the test passes, we don't need this assertion any more.

I will play with test during merge.

Thank you for the update! LGTM